collection/oneidentity/safeguard/plugins/lookup/safeguardpassword.py

```diff
@@ -0,0 +1,233 @@
+            if (
+                not r["WasExpired"]
+                and (  # Query "q" parameter could match by several fields. Match if the asset name or network address is the asset_name passed to the lookup
+                    r["AccountAssetName"].lower() == asset_name.lower()
+                    or r["AssetNetworkAddress"] == asset_name
+                )
+                and (
+                    r["State"] == "RequestAvailable"
+                    or r["State"] == "PasswordCheckedOut"
+                )
+            ):
+                return r["Id"]
+
+        raise AnsibleError(
+            f"Multiple access requests found for '{asset_name}', but no one matches the server (name or ip), not expired and available"
+        )
+
+    @retry(
+        retry=retry_if_exception_type(AnsibleError),
+        stop=stop_after_attempt(4),
+        wait=wait_random_exponential(multiplier=1, min=3, max=30),
+        reraise=True,
+    )
+    def get_password(self, asset_name):
+        # Use query to get the entitlements matching server
+        result = self.connection.invoke(
+            HttpMethods.GET,
+            Services.CORE,
+            "Me/RequestEntitlements",
+            query={
+                # AssetName filter always return an empty list
+                "q": asset_name,
+                "accessRequestType": "Password",
+            },
+        )
+
+        if result.status_code != 200:
+            raise AnsibleError(
+                f"Error obtaining entitlements: {result.status_code} - {result.text}"
+            )
+
+        entitlements = result.json()
+
+        # Filter to get the exact match. Match by name or network address. Ignore case.
+        asset = list(
+            x
+            for x in entitlements
+            if x.get("Account", {}).get("AssetName").lower() == asset_name.lower()
+            or x.get("Account", {}).get("AssetNetworkAddress") == asset_name
+        )
+
+        if not asset:
+            raise AssetNotFoundError(f"Asset with name '{asset_name}' not found")
+        elif len(asset) > 1:
+            raise AnsibleError(
+                f"Multiple entitlements found for '{asset_name}': {asset}"
+            )
+
+        asset = asset[0]
+
+        # Create a request for a password
+        account_id = asset["Account"]["Id"]
+        asset_id = asset["Account"]["AssetId"]
+        result = self.connection.invoke(
+            HttpMethods.POST,
+            Services.CORE,
+            "AccessRequests",
+            body={
+                "AccountId": account_id,
+                "AssetId": asset_id,
+                "AccessRequestType": "Password",
+                "ReasonComment": f"Ansible lookup plugin request for asset '{asset_name}'",
+            },
+        )
+
+        # if error 400, the password is already requested, get the request id
+        if (
+            result.status_code == 400
+            and "You already have a request for the account" in result.text
+        ):
```

Comment:
In this case, you should receive back the following JSON:
```
{
"Code":90001,
"Message":"You already have a request for the account 'someaccount' (37) on asset 'some asset' (29) in the same or overlapping time frame.",
"InnerError":null
}
```
The error messages in Safeguard are localized to different languages and can even be changed between versions. But the error code is fixed and a much more reliable value to check your error condition against.

Can you change your `and` clause to instead compare against the `Code` property of the JSON response for the `90001` value.